### PR TITLE
invoke.mdx: re-written and expanded

### PR DESCRIPTION
**Pull request by audionerd**
Based on:
- https://stately.ai/docs/xstate-v4/category/xstate-actors
- https://xstate.js.org/docs/guides/communication.html

### CODE REVIEW

File: docs/invoke.mdx [hunk 1]
```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
+    failure: {
+      on: {
+        RETRY: { target: 'loading' }
+      }
+    }
+  }
+});
+```
+
+Actors can also be invoked on the _root_ of the machine, where they will last the lifetime of the machine:
```
Comment (davidkpiano) [inline docs/invoke.mdx:77]:
```suggestion
Actors can also be invoked on the _root_ of the machine, and they will be active for the lifetime of its parent machine actor:
```

File: docs/invoke.mdx [hunk 2]
```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
+
+For further examples, see:
+- [Reusing function and event definitions workflow](https://github.com/statelyai/xstate/tree/next/examples/workflow-reusing-functions/main.ts)
+- [Check inbox periodically (cron-based workflow)](https://github.com/statelyai/xstate/tree/next/examples/workflow-check-inbox/main.ts)
+- [Car vitals checks (SubFlow Repeat) workflow](https://github.com/statelyai/xstate/tree/next/examples/workflow-car-vitals)
+
+
+## How are Actors Different From Actions?
+
+Actions are “fire-and-forget”; as soon as their execution starts, the statechart running the actions forgets about them. If you specify an action as `async`, **the action won’t be awaited before moving to the next state**.
```
Comment (davidkpiano) [inline docs/invoke.mdx:126]:
```suggestion
Actions are “fire-and-forget”; as soon as their execution starts, the state machine running the actions forgets about them. If you specify an action as `async`, **the action won’t be awaited before moving to the next state**.
```

File: docs/invoke.mdx [hunk 3]
```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
+
+Invoked actors can do asynchronous work _and_ communicate with their parent machine. They they can send and receive events. Invoked machine actors can even invoke or spawn their own child actors.
```
Comment (davidkpiano) [inline docs/invoke.mdx:128]:
```suggestion
Invoked actors can do asynchronous work _and_ communicate with their parent machine actor. They they can send and receive events. Invoked machine actors can even invoke or spawn their own child actors.
```

File: docs/invoke.mdx [hunk 4]
```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
-  - `id` - The ID of the actor.
-  - `input` - The input to pass to the actor.
-  - `systemId` - system-wide ID
-  - `onDone`
-  - `onSnapshot`
-  - `onError`
+## Lifecycle
 
-## Source
+Invoked actors have a lifecycle that is managed by the parent machine. They are created and started when the state is entered, and stopped when the state is exited.
```
Comment (davidkpiano) [inline docs/invoke.mdx:154]:
```suggestion
Invoked actors have a lifecycle that is managed by the state they are invoked in. They are created and started when the state is entered, and stopped when the state is exited.

If a state is entered and then immediately exited, e.g. due to an eventless ("always") transition, then no actors will be invoked on that state.
```

File: docs/invoke.mdx [hunk 5]
```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
+const machine = createMachine({
+  // …
+  invoke: {
+    src: 'workflow' // string reference
+  }
+});
+
+const actor = createActor(machine.provide({
+  actors: {
+    workflow: fromPromise(…) // provided
```
Comment (davidkpiano) [inline docs/invoke.mdx:225]:
```suggestion
    workflow: fromPromise(/* ... */) // provided
```

File: docs/invoke.mdx [hunk 6]
```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
+    onDone: {/* ... */} // refers to `someSrc` being done
+  },
+  initial: 'loadFoo',
+  states: {
+    loadFoo: {/* ... */},
+    loadBar: {/* ... */},
+    loadingComplete: { type: 'final' }
+  },
+  onDone: 'loaded' // refers to 'loading.loadingComplete' being reached
+}
```
Comment (davidkpiano) [inline docs/invoke.mdx:256]:
```suggestion
```js
// ...
loading: {
  invoke: {
    src: someSrc,
    // refers to `someSrc` being done
    // highlight-next-line
    onDone: {/* ... */}
  },
  initial: 'loadFoo',
  states: {
    loadFoo: {/* ... */},
    loadBar: {/* ... */},
    loadingComplete: { type: 'final' }
  },
  // refers to 'loading.loadingComplete' being reached
  // highlight-next-line
  onDone: { target: 'loaded' }
}
```

**Review by davidkpiano — APPROVED** [review]

File: docs/invoke.mdx (hunk 2 above)
Comment (audionerd) [reply docs/invoke.mdx:126]: Can we define _zero-time_ here? Might not be a familiar term to beginners? Maybe something like: 

> Remember: transitions are always _zero-time_ (states transition synchronously).

Comment (davidkpiano) [reply docs/invoke.mdx:124]: Yes, good idea

File: docs/invoke.mdx [hunk 7]
```diff
@@ -22,71 +18,701 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
+      }
+    },
+    authorized: {
+      type: 'final'
+    }
+  }
+});
+```
+
+That specific example could be simplified by using the `sendParent(...)` action creator for the same effect. But by using `sendTo` the actors do not strictly need to be in a direct parent-child relationship to communicate.
```
[the remark this answers is not in the document]
Comment (davidkpiano) [reply docs/invoke.mdx:675]: Yes

File: docs/invoke.mdx [hunk 8]
```diff
@@ -22,71 +18,701 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
...
 
-- Actors can receive input. This is an event: `{ type: 'xstate.init', input: ... }`
+## Testing
 
-## Kinds of actors
+_Coming soon_
+
+## Referencing Invoked Actors
+
+Actors can be read on `snapshot.children.<actorId>[index]`. The returned value is an `ActorRef` object, with properties like:
```
Comment (audionerd) [inline docs/invoke.mdx:703]: Is there a better way to describe that `snapshot.children.<actorId>` will be an array? Wasn't sure how to word this.

Comment (audionerd) [reply docs/invoke.mdx:701]: Got it. Will fix.

I think I got confused by the string names from a prior XState beta version (was `checkcarvitals.DoCarVitalChecks[0]` vs `0.checkcarvitals.DoCarVitalChecks` or something)

**Review by davidkpiano — APPROVED** [review]

**Pull request merged**